pallets/restricted-tokens/src/lib.rs

```diff
@@ -202,7 +211,7 @@ pub mod pallet {
 			+ LockableCurrency<Self::AccountId>
 			+ ReservableCurrency<Self::AccountId>
 			+ fungible::Inspect<Self::AccountId, Balance = Self::Balance>
-			+ fungible::InspectHold<Self::AccountId, Reason = ()>
+			+ fungible::InspectHold<Self::AccountId, Reason = Self::RuntimeHoldReason>
```

Comment:
⚠️ This will require migration because now the storage is indexed with `HoldReason::NativeIndex` instead of `()`.

I guess it will not be easy to know which amounts must be migrated to this value and which amounts must be migrated to the other variant `HoldReason::TrasnferAllowance` added in `pallet-transfer-alloance`. I think we only have those two possibilities. But not 100% if another pallet adds another variant to `RuntimeHoldReason` that we're using right now. TODO check this

Reply:
We need to ensure that pallets using restricted tokens, which are:
- [ ] pallet-token-mux
- [ ] pallet-liquidity-pools
- [ ] pallet-order-book
- [ ] pallet-block-rewards
- [ ] pallet-rewards (BlockRewards instance)
- [ ] pallet-rewards (Liquidity instance)
- [ ] pallet-investments
- [ ] pallet-restricted-tokens (by extrinsics)
- [ ] pallet-pool-fees
- [ ] pallet-pool-system
- [ ] pallet-fees
- [ ] pallet-treasuty
- [ ] pallet-uniques
- [ ] pallet-vesting
- [ ] pallet-identity
- [ ] pallet-election-phragmen
- [ ] pallet-proxy

have not never used a hold call using `Currency::Native`.

If we can not ensure that, the migration will be very difficult to determine how to split the amount under `()` for each holding purpose of each pallet.